zendesk/zendesk_client_service.go

```diff
@@ -243,19 +243,32 @@ func (c *client) getAll(endpoint string, in interface{}) ([]Ticket, error) {
 	apiStartIndex := strings.Index(dataPerPage.NextPage, apiV2)
 	prevPage := ""
 
-	for dataPerPage.NextPage != prevPage {
-		if fieldName == "tickets" {
-			result = append(result, dataPerPage.Tickets...)
+	for {
+		// if too many requests(res.StatusCode == 429), delay sending request
+		if res.StatusCode == 429 {
+			after, err := strconv.ParseInt(res.Header.Get("Retry-After"), 10, 64)
+			if err != nil {
+				return nil, err
+			}
+			time.Sleep(time.Duration(after) * time.Second)
```

Comment:
I would log here with sleep time.  So if we are debugging we know how long it slept